api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2647,49 +2715,45 @@ public void composeObject(
       }
     }
 
-    if (partsCount == 1) {
+    if (args.sources().size() == 1) {
       ComposeSource src = sources.get(0);
-      if (headerMap == null) {
-        headerMap = new HashMap<>();
-      }
-      if ((src.offset() != null) && (src.length() == null)) {
-        headerMap.put("x-amz-copy-source-range", "bytes=" + src.offset() + "-");
-      }
-
-      if ((src.offset() != null) && (src.length() != null)) {
-        headerMap.put(
-            "x-amz-copy-source-range",
-            "bytes=" + src.offset() + "-" + (src.offset() + src.length() - 1));
+      Multimap<String, String> headers = HashMultimap.create();
+      headers.putAll(args.extraHeaders());
+      headers.putAll(args.headers);
```

Comment:
1. Isn't it `headers.putAll(args.headers())` ? Not sure why there is no compiler error to this line.
2. Why is it required to create combined headers here? Can't you use them directly like `.extraHeaders(args.extraHeaders()).headers(args.headers())` ?